controller/RestDelivery.php

```diff
@@ -212,6 +214,40 @@ public function updateDeferred()
         }
     }
 
+    /**
+     * Delete delivery by URI
+     */
+    public function deleteDeferred()
+    {
+        try {
+            if ($this->getRequestMethod() !== \Request::HTTP_DELETE) {
+                throw new \common_exception_NotImplemented('Only delete method is accepted to updating delivery');
+            }
+
+            $propertyValues = $this->getRequestParameters();
```

Comment:
Please use hasRequestParameter and getRequestParameter instead of getting all of them.
You can check the validation of the uri in the generateDeffered@105